website_sale_phone_validation/controllers/main.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2020 Carlos Lopez Mite <carlos.lopez@odooerp.cl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import tools
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+    def _get_phone_fields_to_validate_sale(self):
+        return ["phone", "mobile"]
+
+    def checkout_form_validate(self, mode, all_form_values, data):
+        partner_model = request.env["res.partner"].sudo()
+        error, error_message = super(WebsiteSale, self).checkout_form_validate(
+            mode, all_form_values, data
+        )
+        for field_name in self._get_phone_fields_to_validate_sale():
+            if not data.get(field_name):
+                continue
+            try:
+                data.update(
+                    {field_name: partner_model.phone_format(data.get(field_name))}
```

Comment:
I think that there is a problem with this line, `res.partner` has no the method `phone_format()`, I tried use it and I get an error, please check it in the following recording:
![Shop _ My Website](https://user-images.githubusercontent.com/334317/89940587-bd6a2580-dbe7-11ea-9332-d1a0f257740c.gif)
 
Can you please describe how to use the module in readme/USAGE.rst ? Also check the travis logs, something is breaking.

Reply:
Hi, yes, method `phone_format `is on model `phone.validation.mixin`, i update code for fix the problem, please try again